src/modals/select-baker-modal/select-baker-modal.tsx

```diff
@@ -169,52 +171,73 @@ export const SelectBakerModal: FC = () => {
-          />
-        </View>
+        {isTezosNode && (
+          <View style={styles.upperContainer}>
+            <Text style={styles.infoText}>The higher the better</Text>
+
+            <Sorter
+              sortValue={sortValue}
+              description="Sort bakers by:"
+              sortFieldsOptions={bakersSortFieldsOptions}
+              sortFieldsLabels={bakersSortFieldsLabels}
+              onSetSortValue={handleSortValueChange}
+              testID={SelectBakerModalSelectors.sortByDropDownButton}
+            />
+          </View>
+        )}
       </View>
 
-      <FlatList
-        data={finalBakersList}
-        renderItem={({ item }) => (
-          <BakerListItem
-            item={item}
-            selected={item.address === selectedBaker?.address}
-            onPress={setSelectedBaker}
-            testID={SelectBakerModalSelectors.bakerItem}
-          />
-        )}
-        keyExtractor={item => item.address}
-        style={styles.flatList}
-        windowSize={10}
-        ListEmptyComponent={
+      {isTezosNode && (
+        <FlatList
+          data={finalBakersList}
+          renderItem={({ item }) => (
+            <BakerListItem
+              item={item}
+              selected={item.address === selectedBaker?.address}
+              onPress={setSelectedBaker}
+              testID={SelectBakerModalSelectors.bakerItem}
+            />
+          )}
+          keyExtractor={item => item.address}
+          style={styles.flatList}
+          windowSize={10}
+          ListEmptyComponent={
+            <BakerListItem
+              item={{ ...emptyBaker, name: UNKNOWN_BAKER_NAME, address: searchValue ?? '', isUnknownBaker: true }}
+              onPress={setSelectedBaker}
+              selected={searchValue === selectedBaker?.address}
+            />
+          }
+        />
+      )}
+
+      {isDcpNode && isValidAddress(searchValue ?? '') && searchValue !== selectedAccount.publicKeyHash && (
+        <View style={styles.dcpBaker}>
+          <Divider size={formatSize(16)} />
           <BakerListItem
             item={{ ...emptyBaker, name: UNKNOWN_BAKER_NAME, address: searchValue ?? '', isUnknownBaker: true }}
             onPress={setSelectedBaker}
             selected={searchValue === selectedBaker?.address}
           />
-        }
-      />
-
-      <ModalButtonsContainer>
-        <ButtonLargeSecondary title="Close" onPress={goBack} testID={SelectBakerModalSelectors.closeButton} />
-        <Divider size={formatSize(16)} />
-        <ButtonLargePrimary
-          title="Next"
-          disabled={!isDefined(selectedBaker) || !isValidAddress(selectedBaker.address)}
-          onPress={handleNextPress}
-          testID={SelectBakerModalSelectors.nextButton}
-        />
-      </ModalButtonsContainer>
+        </View>
+      )}
+
+      {isDcpNode && searchValue === selectedAccount.publicKeyHash && (
+        <Text style={styles.errorText}>You can not delegate to yourself</Text>
```

Comment:
as I understood, it could be some problems with the wallet if we delegate to ourselves. It was not discussed in the task, but we talked about it at the daily meeting.